server/automaton-maven-plugin/pom.xml

```diff
@@ -0,0 +1,46 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <parent>
+        <groupId>org.infinispan</groupId>
+        <artifactId>infinispan-server-parent</artifactId>
+        <version>9.4.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.infinispan</groupId>
+    <artifactId>infinispan-server-automaton-maven-plugin</artifactId>
```

Comment:
Could we name this "infinispan-server-protocol-parser-generator-maven-plugin" and generally avoid the 'automaton' name (which is vague) and call these things 'protocol-parser' ?

Reply:
I don't insist on the current name, but 'protocol-parser' sounds even more vague to me :) I'd like to stress the fact that it uses a state automaton (strictly speaking it's not FSM, though). Maybe `infinispan-server-automaton-protocol-parser-generator-maven-plugin` but I wouldn't want to type that on command line :)

Reply:
Protocol parsing is what it does. The automaton is just an implementation detail, that we do not need to stress, imho.

Reply:
So this generates the server side of the protocol parser. Any chance to extend it to also generate the client side?

Reply:
@anistor https://github.com/infinispan/infinispan/pull/5858#issuecomment-378212962

> I don't think that client code generator would be that useful: writing request down is dead-simple and you know what response is expected. To me it seems that the exception-based approach utilized in ReplayingDecoder works well on client side.

Unless we introduce some very complex responses I don't think it's worth it.

Anyway, if I am renaming this I could omit the `server` and call it `infinispan-protocol-parser-generator-maven-plugin`, putting this out of the server directory.

Reply:
+1 to move it out of the server

I had to enable `SingleByteFrameDecoderChannelInitializer` in all the tests to weed out replay bugs in the server. I see the client tests don't use it, so there may be some replay bugs lurking, even with the responses being simpler than the requests. I'm not particularly fond of `SingleByteFrameDecoderChannelInitializer`, because it generates lots of extra trace logs, so being able to test replay in one specific test and ignore in the others is a big plus of the parser generator for me, for the client and the server.